### PR TITLE
Separate LocalRunner and LocalTFRunner

### PR DESCRIPTION
**Pull request by zequnyu**
- separated tf sessions from LocalRunner to LocalTFRunner
- updated docs and tests correspondingly

### CODE REVIEW

File: src/garage/experiment/local_runner.py [hunk 1]
```diff
@@ -1,69 +1,24 @@
...
 class LocalRunner:
-    """This class implements a local runner for tensorflow algorithms.
-
-    A local runner provides a default tensorflow session using python context.
-    This is useful for those experiment components (e.g. policy) that require a
-    tensorflow session during construction.
-
-    Use Runner.setup(algo, env) to setup algorithm and environement for runner
-    and Runner.train() to start training.
+    """Base class of local runner.
```
Comment (krzentner) [inline src/garage/experiment/local_runner.py]: I feel like this docstring should provide more context of what the runner is for.
Specifically, that the runner is the interface which controls the lifecycle of the algorithm, and manages services within that lifecycle (specifically, snapshotting, resuming, and sampling).

Comment (ryanjulian) [reply src/garage/experiment/local_runner.py]: Yes -- please provide a nice long description. this is a major interface.

File: src/garage/tf/experiment/__init__.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
+"""TF Experiment functions."""
```
Comment (ryanjulian) [inline src/garage/tf/experiment/__init__.py]: No need to abbreviate TensorFlow here. we don't pay for code storage by the character :P

File: src/garage/tf/experiment/local_tf_runner.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+                env=env,
+                policy=policy,
+                baseline=baseline,
+                max_path_length=100,
+                discount=0.99,
+                max_kl_step=0.01)
+            runner.setup(algo, env)
+            runner.train(n_epochs=100, batch_size=4000)
+
+        // to resume immediately.
```
Comment (ryanjulian) [inline src/garage/tf/experiment/local_tf_runner.py]: // cannot be used for comments in python...

**Comment by ryanjulian** [comment]
I'm not sure why the Docker cache is not working in the CI (which is why your build didn't pass). I restarted the job -- ping me if that doesn't fix it. You may also need to rebase on master for the cache to work properly.

**Comment by krzentner** [comment]
I'm merging this, since we're about to delete the Dockerfile.intel image anyways, and the PyTorch folks would benefit from this change.

**Pull request merged**